### ISSUE TITLE
localized case insensitive string comparison is broken

### ISSUE THREAD

**Issue by bradallred** [gemrb/gemrb#1541]
we use `stricmp` to compare strings, which works most of the time since `ResRef`s are ASCII, but `ieVariables` aren't necessarily ASCII (eg PC name), so `stricmp` does not work for those. Additionally, `stricmp` is limited to `char` type, but I don't know that this matters for our use.

`wcsncasecmp` is locale based the same way `towupper` and `towlower` are, which is what we use for changing strings to upper and lower case.

I'm not sure of any specific breakage, but we should be consistent with the rules we are using ([ISO 30112](http://www.open-std.org/JTC1/SC35/WG5/docs/30112d10.pdf))

we probably have a bunch of other problematic functions too, but string comparison is critical and could lead to very peculiar behavior compared to `isdidgit` or `isspace` etc.

**Comment by lynxlynxlynx** [comment]
I very much doubt we use PC names for anything but display, as internally it's all about InParty, global IDs and in the scripting/dialog layer all about scriptnames (death vars).

**Comment by bradallred** [comment]
oh, good. nothing is likely broken then. It's just a question of where we are using such comparisons outside of `ResRef` and `ieVariable`

**Comment by lynxlynxlynx** [comment]
All the other regular char stuff? I think it's easier to ask where non-ASCII text can sneak in, since that's a very limited set.

**Comment by lynxlynxlynx** [comment]
I've checked our `stricmp` uses and the only places where I saw we could have problems are path related (`Interface::SavedExtension`, `Interface::ProtectedExtension`), since dentries use normal chars (not sure how that plays with non-ASCII compatible FS encodings). And definitely `IsQuickSaveSlot`, since we don't limit savegame names to ASCII. And even if we did, users may come with them already from before.
